tests/playwright-test/ui-mode-test-output.spec.ts

```diff
@@ -114,6 +114,45 @@ test('should show console messages for test', async ({ runUITest }, testInfo) =>
   await expect.soft(page.getByText('GREEN', { exact: true })).toHaveCSS('color', 'rgb(0, 188, 0)');
 });
 
+test('should collapse repeated console messages for test', async ({ runUITest }) => {
+  const { page } = await runUITest({
+    'a.spec.ts': `
+      import { test, expect } from '@playwright/test';
+      test('print', async ({ page }) => {
+        await page.evaluate(() => {
+          for (let i = 0; i < 10; ++i)
```

Comment:
For some reason if you add a log immediately before this `for` loop, the first `page message` line will have 20 instances. I haven't looked into why.

Reply:
Good catch! It says "20 page, 10 node, 10 page" instead of "10 page, 10 node, 10 page, 10 page". So the 3rd batch gets packed together with the first one. I think that's just a consequence of the fact that we first write all console messages onto `entries` and then all stdio messages. Don't think it's a big problem - this only occurs because they're so close together.

Reply:
I think ever displaying logs out of order is a regression that will confuse users. I would prefer that we fix this before merging.